restrict artifacts greater than size threshold for trial

anandjakhaniya opened the pull request:
**Jira Ticket:** https://vertaai.atlassian.net/browse/VR-7371

anandjakhaniya committed:
restrict artifacts greater than size threshold for trial

anandjakhaniya committed:
Restrict multipart upload for trial

ravishetye reviewed (commented):
as per a slack discussion we need to pass on ResourceExhausted as the standard error code and the message with a predefined prefix

ravishetye commented on backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/ArtifactStoreService.java:
```diff
@@ -114,4 +116,22 @@ default String getUrl(
+  default void validateArtifactSizeForTrial(App app, int artifactSize, Long partNumber)
+      throws ModelDBException {
+    if (app.getTrialEnabled()) {
+      if (partNumber != null && partNumber != 0) {
```
we should return multipart upload false in trial mode

ravishetye replied:
this probably should be 1 https://github.com/VertaAI/modeldb/blob/master/client/verta/verta/_tracking/experimentrun.py#L272

ravishetye reviewed (changes requested):
this needs update for presigned url flow , I will follow up with you on slack

anandjakhaniya committed:
Merge branch 'aj/config-support-for-trial' into aj/restrict-artifact-size-threshold

anandjakhaniya committed:
POST presigned URL for the trial

anandjakhaniya committed:
Merge branch 'aj/config-support-for-trial' into aj/restrict-artifact-size-threshold

anandjakhaniya committed:
null check for absent value on trial config

ravishetye commented on backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java:
```diff
@@ -2410,7 +2428,13 @@ public void commitMultipartArtifact(
+        throw new ModelDBException(
+            "Multipart artifact upload not supported on the trial version",
+            Code.FAILED_PRECONDITION);
       }
```
Follow the format for errot messages and error code

anandjakhaniya replied:
it is already resolved on other subsequent PR https://github.com/VertaAI/modeldb/pull/1616

ravishetye commented:
When we fetch the temporary credentials let us save them and reuse for this request

anandjakhaniya committed:
Fix logic for the temporary credentials

anandjakhaniya committed:
hard code maxArtifactSizeMB to 2MB for testing

anandjakhaniya committed:
trial config hard coded for testing

anandjakhaniya committed:
revert hard coded config

anandjakhaniya committed:
set default MAX_ARTIFACT_SIZE_DEFAULT to 1GB

Commit:
debug

ravishetye commented on backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Service.java:
```diff
@@ -124,6 +124,52 @@ private Boolean doesObjectExist(String bucketName, String path) throws ModelDBEx
+  public GetUrlForArtifact.Response generatePresignedUrlForTrial(
+      String s3Key, String method, long partNumber, String uploadId) throws ModelDBException {
+
+    if (partNumber != 0) {
```
this should change to 1 based on the way client checks things
https://github.com/VertaAI/modeldb/blob/master/client/verta/verta/_tracking/experimentrun.py#L272

anandjakhaniya committed:
Set credentials object instead of accessKey and secret key

ravishetye commented on backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/ArtifactStoreService.java:
```diff
@@ -114,4 +120,27 @@ default String getUrl(
+          && uploadedArtifactSize > ((double) app.getMaxArtifactSizeMB() * 1024)) {
+        throw new ModelDBException(
+            ModelDBConstants.LIMIT_RUN_ARTIFACT_SIZE
+                + "Artifact size more then "
```
```suggestion
                + "Artifact size more than "
```

ravishetye commented on backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/ArtifactStoreService.java:
```diff
@@ -114,4 +120,27 @@ default String getUrl(
+            ModelDBConstants.LIMIT_RUN_ARTIFACT_SIZE
+                + "Artifact size more then "
+                + app.getMaxArtifactSizeMB()
+                + " MB not supported",
```
```suggestion
                + " MB not supported during the trial",
```

ravishetye commented on backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/ArtifactStoreService.java:
```diff
@@ -114,4 +120,27 @@ default String getUrl(
+      if (partNumber != null && partNumber != 0) {
+        throw new ModelDBException(
+            ModelDBConstants.LIMIT_RUN_ARTIFACT_SIZE
+                + "Multipart artifact upload not supported on the trial version",
```
```suggestion
                + "Multipart artifact upload not supported during the trial",
```

ravishetye commented on backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Client.java:
```diff
@@ -69,6 +78,19 @@ public S3Client(String cloudBucketName) throws IOException {
+  private void initializeS3SignatureUtil() throws ModelDBException {
+    if (app.getTrialEnabled()) {
+      if (awsCredentials == null) {
+        throw new ModelDBException("awsCredentials should be required for the verta trial");
```
```suggestion
        throw new ModelDBException("awsCredentials are required for the verta trial");
```

ravishetye commented on backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Service.java:
```diff
@@ -124,6 +124,52 @@ private Boolean doesObjectExist(String bucketName, String path) throws ModelDBEx
+    if (partNumber != 0) {
+      throw new ModelDBException(
+          ModelDBConstants.LIMIT_RUN_ARTIFACT_SIZE
+              + "Multipart artifact upload not supported on the trial version",
```
```suggestion
              + "Multipart artifact upload not supported during the trial",
```

ravishetye commented on backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Service.java:
```diff
@@ -124,6 +124,52 @@ private Boolean doesObjectExist(String bucketName, String path) throws ModelDBEx
+            ModelDBConstants.LIMIT_RUN_ARTIFACT_SIZE
+                + "Method type ("
+                + method
+                + ") not supported on the trial",
```
```suggestion
            ModelDBConstants.LIMIT_RUN_ARTIFACT_SIZE
                + "Method type "
                + method
                + " is not supported during the trial",
```

anandjakhaniya committed:
Fix review comments

anandjakhaniya committed:
Revert part number suggestion

anandjakhaniya committed:
Merge branch 'aj/config-support-for-trial' into aj/restrict-artifact-size-threshold

# Conflicts:
#	backend/src/main/java/ai/verta/modeldb/ModelDBConstants.java
#	backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunServiceImpl.java

anandjakhaniya committed:
Fix error message as per design doc
Throw 400 when client called PUT request for the PresignedUrl on the trial mode
